src/python/pants/backend/javascript/subsystems/nodejs.py

```diff
@@ -41,6 +45,26 @@ class NodeJS(TemplatedExternalTool):
         "linux_x86_64": "linux-x64",
     }
 
+    resolves = DictOption[str](
+        default={},
+        help=softwrap(
+            f"""
+            A mapping of names to lockfile paths used in your project.
+
+            The default name of a resolve will be the path to the directory containing
+            a lockfile, from the build root, {os.path.sep} replaced with '.'.
```

Comment:
```suggestion
            Specifying a resolve name for each `{PackageJsonTarget.alias}`
            is optional. If unspecified, the default resolve name is calculated by
            taking the path from the build root to the directory containing
            the lockfile and replacing '{os.path.sep}' with '.' in that path.
```

And add the import: `from pants.backend.javascript.package_json import PackageJsonTarget`

Reply:
So, what is the definition of a resolve for nodejs then? Does the lockfile have to exist for something to be considered a resolve? So, if someone hasn't created a lockfile yet, then a package.json file is not part of a resolve? When does a directory get included as the root of a resolve?

Reply:
A node resolve, concretely, is:
1. npm/yarn: A package.json that is not pointed to by other `package.json#workspaces`, or
2. pnpm: Either a single package.json, or a root directory containing a `pnpm-workspace.yaml`.
All cases share that they will install their corresponding lockfile at the _root_ directory of the "resolve".

pnpm is really what is making this less clear cut, but the intention is to support all of them.

Reply:
```suggestion
            Specifying a resolve name is optional. If unspecified,
            the default resolve name is calculated by taking the path
            from the build root to the directory containing the lockfile
            and replacing '{os.path.sep}' with '.' in that path.
```

Reply:
Ok. Thanks for helping me understand a bit better. I have an alternate suggestion that doesn't mention `package_json`.